Excel input enabled with --xlsx and Required values made configurable…

somanath304 opened the pull request:
… in example_config.py file

somanath304 committed:
Excel input enabled with --xlsx and Required values made configurable in example_config.py file

somanath304 committed:
Hard coded excel file path removed and made dynamic

kerchner commented:
@somanath304 Thanks for this Pull Request.  Now that the code requires the `xlsx` library, you'll need to create a `requirements.txt` file.  Setting the code up will require creating a python virtual environment (such as with virtualenv or venv) and then `pip install -r requirements.txt`.  I'm still going through the code but you can commit that change to this branch, and it will become part of the pull request.

kerchner commented:
@somanath304 Regarding `example_config.py` - It looks like the example fields `ingest_path`, `ingest_command` and `debug_mode` dropped out.  I think we need both the example fields that were there, plus the new ones you added.  The user should be able to copy `example_config.py` to `config.py` and then run the example with no changes necessary.

somanath304 committed:
Create config.py

Adding the config.py file which contains the ingest_path, ingest_command and debug mode.

somanath304 committed:
Update example_config.py

This file contains the required values , to add new values append the field name at the end of the list

somanath304 committed:
Config file - contains ingest_path , ingest_command and debug mode control

somanath304 committed:
Update batch_loader.py

somanath304 committed:
Delete example_config.py

somanath304 committed:
Update config.py

somanath304 commented:
> @somanath304 Regarding `example_config.py` - It looks like the example fields `ingest_path`, `ingest_command` and `debug_mode` dropped out. I think we need both the example fields that were there, plus the new ones you added. The user should be able to copy `example_config.py` to `config.py` and then run the example with no changes necessary.

@kerchner - I have deleted the example_config.py file and added the configurable "required values" into config.py file . Now there need not be two config files but just the one which contains all the old data(i.e.ingest_path,ingest_command and debug mode) plus the new data which was added(i.e.required values)

somanath304 committed:
Added requirements.txt file

Added requirements.txt file

somanath304 commented:
> @somanath304 Thanks for this Pull Request. Now that the code requires the `xlsx` library, you'll need to create a `requirements.txt` file. Setting the code up will require creating a python virtual environment (such as with virtualenv or venv) and then `pip install -r requirements.txt`. I'm still going through the code but you can commit that change to this branch, and it will become part of the pull request.

@kerchner Added the requirements.txt file to this branch

kerchner commented:
@somanath304 thanks.  A few things:
- Please move `config.py` back to `example.config.py` as it is intended as an example.
- Update README.md documentation to specify how to use the new `--xlsx` option
- Update README.md documentation to specify how to list the required fields, now that this is configurable
- In the config file, is there a reason that the list of fields is a set, inside `{}` rather than a tuple, inside `()` ?
- In batch_loader.py near the top, where you set required_field_names to cfg1.required, is there a reason that the latter is inside parentheses?
- What is accomplished by giving a nickname cfg1 for config? (`import config as cfg1`) .  It looks like you only need to reference it once -- but "config" is used several times later.

Commit:
required values have been moved to example.config.py file

somanath304 committed:
Delete config.py

somanath304 committed:
Update README.md

somanath304 commented:
> @somanath304 thanks. A few things:
> 
> * Please move `config.py` back to `example.config.py` as it is intended as an example.
> * Update README.md documentation to specify how to use the new `--xlsx` option
> * Update README.md documentation to specify how to list the required fields, now that this is configurable
> * In the config file, is there a reason that the list of fields is a set, inside `{}` rather than a tuple, inside `()` ?
> * In batch_loader.py near the top, where you set required_field_names to cfg1.required, is there a reason that the latter is inside parentheses?
> * What is accomplished by giving a nickname cfg1 for config? (`import config as cfg1`) .  It looks like you only need to reference it once -- but "config" is used several times later.

@kerchner Please find the comments inline 
>* Please move `config.py` back to `example.config.py` as it is intended as an example.- **Moved config to example.config**
 >* Update README.md documentation to specify how to use the new `--xlsx` option- **Instructions on how to run with excel input added**
>* Update README.md documentation to specify how to list the required fields, now that this is configurable- **Added a section with instructions** 
>* In the config file, is there a reason that the list of fields is a set, inside `{}` rather than a tuple, inside `()` ? **Fixed**
> * In batch_loader.py near the top, where you set required_field_names to cfg1.required, is there a reason that the latter is inside parentheses? **Removed cgf1 ,Fixed**
> * What is accomplished by giving a nickname cfg1 for config? (`import config as cfg1`) .  It looks like you only need to reference it once -- but "config" is used several times later **Fixed**
